envctl/envctl

```diff
@@ -4,7 +4,9 @@ set -o pipefail # any step in pipe caused failure
 set -u # undefined variables cause exit
 
 # set an ID to use for future runs
-ENVCTL_ID="${ENVCTL_ID:-$(hostname)}"
+# remove dots from the hostname if present
+HOST=$(echo $(hostname) | sed 's/[^a-zA-Z0-9]//g')
+ENVCTL_ID="${ENVCTL_ID:-$HOST}"
```

Comment:
don't we need uuid to add randomness into the fold?

Reply:
No, we want it to be consistent per-host by default - every time I run a test on my laptop, I want it to test against existing resources when possible, instead of spinning up a new cluster

The CI scripts will overwrite this value with a uuid to make sure they stay random though

Reply:
From framework design perspective, I think the randomness should be handled by envctl instead of by Cis using it. This makes sure the consumers of the framework don't need to make hidden assumptions about how to run the tool properly.

Reply:
The problem is we *don't* want randomness during regular development. When I'm adding a test, I want it to re-use standard test cluster each time, rather than creating a new one for each test run. CI is really the special case where we really do want it to be unique for each run.

We could have it random by default and instruct contributors users to override it with a static value for development, but that's a lot more overhead than setting it to a random value within the CI scripts